[jit] kill isSameIdentity

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35747 [jit] make node hashing use ivalue equality
* **#35019 [jit] kill isSameIdentity**
* #35005 [jit] kill `shallowEquals`
* #34986 [jit] define equality for IValue

Differential Revision: [D20537903](https://our.internmc.facebook.com/intern/diff/D20537903)

suo committed:
[jit] kill isSameIdentity

[ghstack-poisoned]

suo committed:
Update on "[jit] kill isSameIdentity"




[ghstack-poisoned]

suo committed:
Update on "[jit] kill isSameIdentity"




[ghstack-poisoned]

zdevito reviewed (approved):
ok, I see that isSameIdentity is dying.

suo committed:
Update on "[jit] kill isSameIdentity"


Differential Revision: [D20537903](https://our.internmc.facebook.com/intern/diff/D20537903)

[ghstack-poisoned]

suo committed:
Update on "[jit] kill isSameIdentity"


Differential Revision: [D20537903](https://our.internmc.facebook.com/intern/diff/D20537903)

[ghstack-poisoned]

suo committed:
Update on "[jit] kill isSameIdentity"


Differential Revision: [D20537903](https://our.internmc.facebook.com/intern/diff/D20537903)

[ghstack-poisoned]
